### PR TITLE
Extend exception message

### PR DESCRIPTION
**Pull request by orozcodiaz**
Update exception messages to provide more detailed information on errors

### CODE REVIEW

**Review by edenduong — CHANGES_REQUESTED** [review]
Could you please update the `.csv` file?

**Comment by orozcodiaz** [comment]
> Could you please update the `.csv` file?

done

**Review by edenduong — APPROVED** [review]
Thank you for your contribution !

**Comment by magento-engcom-team** [comment]
Hi @edenduong, thank you for the review.
[ENGCOM-6678](https://jira.corp.magento.com/browse/ENGCOM-6678) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
I believe this could be tested with unit test.
@oroskodias will you be able to cover your changes with unit test?

**Comment by engcom-Echo** [comment]
I will take care of test coverage.

File: app/code/Magento/Multishipping/Test/Unit/Model/Checkout/Type/MultishippingTest.php [hunk 1]
```diff
@@ -313,10 +330,13 @@ public function testSetShippingItemsInformation()
     }
 
     /**
-     * @expectedException \Magento\Framework\Exception\LocalizedException
+     * Verefi set shipping items information for address leak
```
Comment (Nazar65) [inline app/code/Magento/Multishipping/Test/Unit/Model/Checkout/Type/MultishippingTest.php]: Verify

File: app/code/Magento/Multishipping/Test/Unit/Model/Checkout/Type/MultishippingTest.php [hunk 2]
```diff
@@ -801,7 +983,12 @@ private function getShippingAssignmentMock()
         return $shippingAssignmentMock;
     }
 
-    private function mockShippingAssignment()
+    /**
+     * Expected shipping assignment
+     *
+     * @return void
+     */
+    private function mockShippingAssignment():void
```
Comment (Nazar65) [inline app/code/Magento/Multishipping/Test/Unit/Model/Checkout/Type/MultishippingTest.php]:
```suggestion
    private function mockShippingAssignment(): void
```

File: app/code/Magento/Multishipping/Test/Unit/Model/Checkout/Type/MultishippingTest.php [hunk 3]
```diff
@@ -530,6 +583,117 @@ public function testCreateOrders(): void
...
+        $simpleProductTypeMock = $this->getMockBuilder(Simple::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getOrderOptions'])
+            ->getMock();
+        $productMock = $this->getProductMock($simpleProductTypeMock);
+        $simpleProductTypeMock->method('getOrderOptions')
+            ->with($productMock)
+            ->willReturn($config['infoBuyRequest']);
+        $quoteItemMock = $this->getQuoteItemMock($config['productType'], $productMock);
+        $quoteAddressItemMock =
```
Comment (Nazar65) [inline app/code/Magento/Multishipping/Test/Unit/Model/Checkout/Type/MultishippingTest.php]:
```suggestion
 $quoteAddressItemMock = $this->getQuoteAddressItemMock(
                                                $quoteItemMock,
                                                $config['productType'], 
                                                $config['infoBuyRequest']
                                                );
```

**Review by Nazar65 — CHANGES_REQUESTED** [review]

**Review by Nazar65 — APPROVED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6678](https://jira.corp.magento.com/browse/ENGCOM-6678) has been created to process this Pull Request

**Review by edenduong — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @edenduong, thank you for the review.
[ENGCOM-6678](https://jira.corp.magento.com/browse/ENGCOM-6678) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Before:**
before

**After:**
after

**Pull request merged**